### PR TITLE
docs(rule-engine): adjust the rule engine document structure

### PR DESCRIPTION
**Pull request by sdzx-1**

### CODE REVIEW

**Comment by zmstone** [comment]
please ensue english version too

File: zh_CN/rule/rule-engine_buildin_function.md [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+
+###`subbits`
+| 函数名 | 函数功能| 参数| 返回值 |
+| ----- | ------ | -- | ----- |
+|subbits| 从二进制数据的起始位置获取指定长度的比特位, 然后转换为无符号整型 (大端). | 1. 二进制数据 <br />2. 要获取的长度(bits) | 无|subbits符号整数 |
+|subbits| 从二进制数据的指定下标位置获取指定长度的比特位, 然后转换为无符号整型 (大端). 下标是从 1 开始的 | 1. 二进制数据 <br />2. 起|subbits始位置的下标 <br />3. 要获取的长度(bits) | 无符号整数 |
+|subbits| 从二进制数据的指定下标位置获取指定长度的比特位, 然后按照给定的参数转换为想要的数据类型. 下标是从 1 开始的. | 1. 二进制数据 <br />2. 起始位置的下标 <br />3. 要获取的长度(bits) <br />4. 数据类型，可选值：'integer', 'float', 'bits' <br />5. 符号类型, 只对整型数据有效, 可选值：'unsigned', 'signed', <br />6. 大端还是小端, 只对整型数据有效, 可选值：'big', 'little' | 获取到的数据 |
+
+```erlang
+subbits('aaa', 10) = 389
```
Comment (terry-xiaoyu) [inline zh_CN/rule/rule-engine_buildin_function.md]: We should use 8 (or Multiple of 8) as the `length` parameter, as the binary to be parse is a string. Like:

```
subbits('aaa', 8) = 97
```

File: zh_CN/rule/rule-engine_buildin_function.md [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+subbits('aaa', 5, 10) = 88
+subbits('aaa', 7, 10, 'integer', 'signed', 'big') = 353
```
Comment (terry-xiaoyu) [inline zh_CN/rule/rule-engine_buildin_function.md]: Add more examples for `subbits` function, using different options.

File: zh_CN/rule/rule-engine_field.md [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+keywords:
+# 描述
+description:
+# 分类
+category:
+# 引用
+ref:
+---
+
+# SELECT 和 WHERE 子句可用的字段
```
Comment (terry-xiaoyu) [inline zh_CN/rule/rule-engine_field.md]: 规则 SQL 语句中可用的字段

File: zh_CN/rule/rule-engine.md [hunk 4]
```diff
@@ -107,1255 +122,4 @@ SELECT * FROM "$events/client_connected"
 Dashboard 中提供了旧版 SQL 语法转换功能可以完成 SQL 升级迁移。
 :::
 
-## 规则引擎组成
```
Comment (terry-xiaoyu) [inline zh_CN/rule/rule-engine.md:110]: Where is this section?

File: zh_CN/rule/rule-engine_grammar_and_examples.md [hunk 5]
```diff
@@ -0,0 +1,510 @@
...
+  "payload": {
+    "data": [
+      {"name": "STEVE", "id": 1},
+      {"name": "bill", "id": 2}
+    ]
+  }
+}
+```
+
+## 事件和事件主题
```
Comment (terry-xiaoyu) [inline zh_CN/rule/rule-engine_grammar_and_examples.md]: Move this section into `rule-engine_fields.md`?

File: zh_CN/rule/rule-engine_grammar_and_examples.md [hunk 6]
```diff
@@ -0,0 +1,510 @@
...
+规则引擎的 SQL 语句既可以处理消息(消息发布)，也可以处理事件(客户端上下线、客户端订阅等)。对于消息，FROM 子句后面直接跟主题名；对于事件，FROM 子句后面跟事件主题。
+
+事件消息的主题以 `"$events/"` 开头，比如 `"$events/client_connected",` `"$events/session_subscribed"。`
+如果想让 emqx 将事件消息发布出来，可以在 `emqx_rule_engine.conf` 文件中配置。
+
+所有支持的事件及其可用字段详见: [规则事件](#rule-sql-events)。
+
+
+### FROM 子句可用的事件主题
```
Comment (terry-xiaoyu) [inline zh_CN/rule/rule-engine_grammar_and_examples.md]: Move this section into `rule-engine_fields.md`?

File: zh_CN/rule/rule-engine.md (hunk 4 above)
Comment (sdzx-1) [reply zh_CN/rule/rule-engine.md:110]: I removed this part. It is an overview of rule-engine, resource, action.

File: en_US/rule/rule-engine_field.md [hunk 7]
```diff
@@ -0,0 +1,399 @@
...
+  "qos": 1,
+  "node": "emqx@127.0.0.1"
+}
+```
+
+## $events/delivery_dropped
+
+Trigger rule when subscriber's message queue is full
+
+### $events/delivery_dropped
```
Comment (terry-xiaoyu) [inline en_US/rule/rule-engine_field.md]: is this line duplicated?

File: en_US/rule/rule-engine_field.md [hunk 8]
```diff
@@ -0,0 +1,399 @@
...
+  "qos": 1,
+  "from_username": "u_emqx_1",
+  "from_clientid": "c_emqx_1"
+}
+```
+## $events/client_connected
+
+Trigger the rule when the terminal is connected successfully
+
+### $events/client_connected
```
Comment (terry-xiaoyu) [inline en_US/rule/rule-engine_field.md]: is this line duplicated?

**Review by terry-xiaoyu — DISMISSED** [review]

**Review by terry-xiaoyu — APPROVED** [review]

**Comment by zmstone** [comment]
should we change 4.3 too ?
if yes, future such PRs should be sent to release-4.3 and merged to release-4.4

**Pull request merged**